activerecord/test/cases/adapters/mysql2/mysql2_adapter_test.rb

```diff
@@ -19,6 +19,22 @@ def test_exec_query_nothing_raises_with_no_result_queries
     end
   end
 
+  def test_database_exists_returns_false_if_database_does_not_exist
+    error = ->(_) { raise Mysql2::Error.new("Unknown database")  }
+    config = ActiveRecord::Base.configurations["arunit"]
+
+    Mysql2::Client.stub(:new, error) do
```

Comment:
This makes me a little 😢, when first attempting to connect to a DB the MySQL will raise a unauthorized user error even if the database does not exist: `Mysql2::Error::ConnectionError: Access denied for user 'rails'@'%' to database 'non_extant_database'`. 

The options I saw was to grant the `rails` user permission to the non extant database and then the test works as expected. However instead of managing permissions from our tests I opted for stubbing, it's not ideal but seems to be the best option, I'm open to ideas!

Reply:
Probably we can use `inexistent_activerecord_unittest` database for that test.